report/frontend-development.md

```diff
@@ -1,18 +1,173 @@
+By leveraging cookies for token storage and refresh management, we provide a secure and user-friendly authentication experience.
+#####  Input Validation and Sanitation
+We perform input validation and sanitization in the frontend to protect our backend from malicious inputs and to catch invalid data early in the process. This proactive approach enhances security while improving the user experience by providing immediate feedback if the input does not meet our predefined standards. For instance, we enforce specific requirements for fields such as passwords and usernames, ensuring they comply with our security and usability guidelines before they are submitted to the server.
+#####  Secure Communication (HTTPS)
+Our frontend exclusively uses HTTPS to ensure that all data transmitted between the client and the server is encrypted. This protects sensitive information, such as authentication tokens and user data, from interception or tampering during transmission, maintaining the integrity and confidentiality of the communication.
 
 #### Package Management
+In our project, we utilize npm (Node Package Manager) as the primary package manager for frontend development. npm is a powerful tool that simplifies the process of managing dependencies and libraries required for the project. By specifying the necessary packages in the package.json file, npm allows us to efficiently install, update, and manage all dependencies in a structured and repeatable manner.
+
+One of the key advantages of npm is its ability to define and execute custom scripts. These scripts streamline various development and build tasks. For example:
+
+npm run dev: This command is configured to start a development server, providing a live-reloading environment that makes it easier to test and iterate on frontend code during development.
+npm run build: This command specifies the steps required to build the production-ready version of the frontend application. It typically compiles, minifies, and optimizes the code for deployment.
+Additionally, npm enhances the security of our project by providing the npm audit tool. This tool scans the project dependencies for known vulnerabilities and security threats. By running npm audit, we can:
+
+ * Detect security issues in real-time across all installed libraries.
+* Receive detailed reports on the nature of vulnerabilities, their severity, and potential fixes.
+* Apply automated patches for certain vulnerabilities using npm audit fix.
+
+By regularly auditing our dependencies, we ensure that our project remains secure and compliant with best practices, reducing the risk of threats from third-party libraries.
+
+In addition to managing dependencies and automating tasks, npm also enables us to run custom test suites. For instance, by specifying test commands in package.json, such as npm run test, we can execute different testing frameworks or tools to ensure the quality and stability of our code.
```

Comment:
To my knowledge we have no tests in the frontend, so mention that this is a future improvement that we can use.